### ISSUE TITLE
Don't send Hint/Response Updated messages for already solved puzzles

### ISSUE THREAD

**Issue by royleban** [PuzzleServer/mainpuzzleserver#427]
Is there ever a case where a team cares about an updated Hint or Response when they've already solved a puzzle? I don't think so and it's a distraction (especially after the event is over!). So, we should leave out of the recipients list any players on teams that have already solved the puzzle.

**Comment by Vroo** [comment]
In the past, there have been puzzles where players needed information from a puzzle after it was solved. In fact, we did this in Puzzlehunt 1. Not knowing if this is the case for any particular puzzle, we should not suppress mail about updates during the hunt.

Suppressing updates after the hunt is over is a good idea.

**Comment by royleban** [comment]
Is there an example of such a situation where it wouldn't be covered by errata? It is already the case that you cannot see the response for a value you didn't submit before the puzzle was solved, nor can you a hint you didn't ask for before the puzzle was solved. Relying on those being able to be updated post-solve when some people can't ever see them is a bad thing.

I think the right modification is this:

Don't send Hint/Response Updated messages for anything _except an isSolved response_ for already solved puzzles _(and suppress all updates after the hunt is over)_

**Comment by tabascq** [comment]
If the update is due to my team's experience, I want to know that it was handled and how. And nobody should be updating these things after the event is over.

**Comment by Vroo** [comment]
I agree we don't need it now. If an error is in a response, then I cannot do that as a regular errata. The correct way to do that is edit the response. And yes, teams might need to know that a response changed even after they solved the puzzle. Example: `Correct! For the meta you will need to use the incorrect spelling REFERER, not REFERRER`.
